airflow/cli/cli_parser.py

```diff
@@ -1779,13 +1786,23 @@ class GroupCommand(NamedTuple):
         name='check',
         help="Checks if job(s) are still alive",
         func=lazy_load_command('airflow.cli.commands.jobs_command.check'),
-        args=(ARG_JOB_TYPE_FILTER, ARG_JOB_HOSTNAME_FILTER, ARG_JOB_LIMIT, ARG_ALLOW_MULTIPLE),
+        args=(
+            ARG_JOB_TYPE_FILTER,
+            ARG_JOB_HOSTNAME_FILTER,
+            ARG_JOB_HOSTNAME_CALLABLE_FILTER,
+            ARG_JOB_LIMIT,
+            ARG_ALLOW_MULTIPLE,
+        ),
         epilog=(
             'examples:\n'
             'To check if the local scheduler is still working properly, run:\n'
             '\n'
             '    $ airflow jobs check --job-type SchedulerJob --hostname "$(hostname)"\n'
             '\n'
+            'To check if the local scheduler is still working properly with `hostname_callback`, run:\n'
+            '\n'
+            '    $ airflow jobs check --job-type SchedulerJob --local\n'
+            '\n'
```

Comment:
```suggestion
```

I think the single example is enough.

However, if we wanted to, we could add a hostname example for a different host.